src/transports/http.ts

```diff
@@ -729,15 +732,19 @@ class TransportHTTP implements TransportService {
       headers,
     };
 
-    const response = await global.fetch(url, httpOptions);
-
-    if (response.status >= 400) {
-      const Err = new Error(response.statusText)
-      logger.error(Err)
-      throw Err
+    try {
+      const response = await axios(url, httpOptions)
+      return response.data
+    } catch (error: any) {
+      if (error.response) {
+        const {status, data} = error.response
+        console.error(`Status: ${status}, Message: ${typeof data === 'string' ? data : data.message }`)
+      } else if (error.request) {
+        console.error(error.request);
+      } else {
+        throw error.message
```

Comment:
When you re-throw the error (in the last else block), you're only throwing the error message, not the entire error object. This might result in loss of error context (like stack trace). Consider re-throwing the entire error object unless you have a specific reason to only throw the message.

Reply:
You're right. Changed.